source/Octopus.Tentacle/Kubernetes/KubernetesPodMonitor.cs

```diff
@@ -85,7 +103,7 @@ async Task UpdateLoop(CancellationToken cancellationToken)
 
         internal async Task<string> InitialLoadAsync(CancellationToken cancellationToken)
         {
-            log.Verbose("Preloading pod statuses");
+            log.Verbose("Loading pod statuses");
 
```

Comment:
Are we missing an `initialLoadLock.Reset()` here? won't the next time the initial load kicks in, the lock will already be set and thus won't block anything?

Reply:
I initially had a reset - but I think we are ok on subsequent loads - I created another dictionary to hold the pending Pods, so `podStatusLookup` is only responsible for holding what comes from the K8S API

Reply:
and I guess we are doing an atomic swap of that dictionary 👍